pkg/aws/common_mutator.go

```diff
@@ -27,6 +28,28 @@ type Mutator struct {
 	Logger    micrologger.Logger
 }
 
+func MutateLabelFromAWSCluster(m *Mutator, meta metav1.Object, awsCluster infrastructurev1alpha2.AWSCluster, label string) ([]mutator.PatchOperation, error) {
+	var result []mutator.PatchOperation
+
+	if meta.GetLabels()[label] != "" {
+		return result, nil
+	}
+
+	// Extract release from Cluster.
+	value := awsCluster.GetLabels()[label]
+	if value == "" {
+		return nil, microerror.Maskf(notFoundError, "AWSCluster %s did not have the label %s set.", awsCluster.GetName(), label)
+	}
```

Comment:
Just sharing something FYI. A map can have a a key and an empty value associated with it. So there is a difference between something not being there and something being an empty string. Not sure what you want to assert but things like these should be accurate. See also https://play.golang.org/p/YJ3i9V8nQSx.

Reply:
Thank you for sharing. That is true.
Here we try to fetch the version number from the awscluster in order to default it. We would get the same error in both cases (the label not existing at all and the label existing but no version number assigned to it) which I thought is fine.
Or do we have a use case for an empty version number value?